libsplinter/src/network/connection_manager/mod.rs

```diff
@@ -604,6 +668,23 @@ fn send_heartbeats<T: MatrixLifeCycle, U: MatrixSender>(
                     }
                 }
             }
+            // We leave it up to the remote to peform heartbeats
+            ConnectionMetadata::Inbound { id, .. } => {
+                info!("Sending heartbeat to {}", endpoint);
+                if let Err(err) = state.matrix_sender().send(id, heartbeat_message.clone()) {
+                    error!(
+                        "failed to send heartbeat: {:?} attempting reconnection",
+                        err
+                    );
+
+                    notify_subscribers(
```

Comment:
So is the idea here that the PeerManager will handle inbound peers differently when it comes to trying different endpoints? The PeerManager currently checks the number of re-connection attempts, and after a specified # of failures tries different endpoints. But these notifications are not sent for inbound connections. 

Also right now this will send a disconnected notification every time a heartbeat fails, (every 10 seconds by default).

Reply:
I've updated the inbound metadata to include a disconnected state - this allows for:
- a heartbeat to fail intermittently
- the layer is notified when the connection is initially detected as disconnected - it's still up to the listening layer to decide what to do at that point.